### PR TITLE
[libc] Move EOF macro to stdio-macros.h

### PR DESCRIPTION
**Pull request by petrhosek**
libc++ char_traits.h assumes EOF is always available

See #85158 for more details.

### CODE REVIEW

File: libc/config/baremetal/api.td [hunk 1]
```diff
@@ -52,6 +52,15 @@ def IntTypesAPI : PublicAPI<"inttypes.h"> {
   let Types = ["imaxdiv_t"];
 }
 
+def StdIOAPI : PublicAPI<"stdio.h"> {
+  let Macros = [
+    SimpleMacroDef<"EOF", "-1">,
```
Comment (lntue) [inline libc/config/baremetal/api.td:57]: Can you add this definition in `libc/include/llvm-libc-macros/stdio-macros.h` instead?

Comment (petrhosek) [reply libc/config/baremetal/api.td]: Done

**Review by lntue — APPROVED** [review]

**Review by lntue — APPROVED** [review]

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Review by SchrodingerZhu — APPROVED** [review]
LGTM

**Pull request merged**